docs(all): TFHE-rs v0.3.0 doc update (#409)
zama-ai/tfhe-rs: tfhe/docs/how_to/js_on_wasm_api.md

    CompactFheUint32List
} = require("./pkg/tfhe.js");

function fhe_uint32_example() {
    // Makes it so that if a rust thread panics,
    // the error message will be displayed in the console
    init_panic_hook();

    const block_params = new ShortintParameters(ShortintParametersName.PARAM_SMALL_MESSAGE_2_CARRY_2_COMPACT_PK);
    let config = TfheConfigBuilder.all_disabled()
        .enable_default_integers()
        .build();

    let clientKey = TfheClientKey.generate(config);
    let compressedServerKey = TfheCompressedServerKey.new(clientKey);
    let publicKey = TfheCompactPublicKey.new(clientKey);

    let values = [0, 1, 2394, U32_MAX];
    let compact_list = CompactFheUint32List.encrypt_with_compact_public_key(values, publicKey);

    let serialized_list = compact_list.serialize();
    let deserialized_list = CompactFheUint32List.deserialize(serialized_list);
    let encrypted_list = deserialized_list.expand();
    assert.deepStrictEqual(encrypted_list.length, values.length);

    for (let i = 0; i < values.length; i++)
    {
        let decrypted = encrypted_list[i].decrypt(clientKey);
        assert.deepStrictEqual(decrypted, values[i]);
    }
}
```

### Web

> - When using the Web WASM target, there is an addition `init` function to

Reviewer: ```suggestion
- When using the Web WASM target, there is an additional `init` function to
```

call.
- When using the Web WASM target with parallelism enabled, there is also
  one more initialization function to call `initThreadPool`

#### Example

```rust
import init, {
    initThreadPool, // only available with parallelism
    init_panic_hook,
    ShortintParametersName,
    ShortintParameters,
    TfheClientKey,
    TfhePublicKey,
} from "./pkg/tfhe.js";

async function example() {
    await init()
    await initThreadPool(navigator.hardwareConcurrency);
    await init_panic_hook();

    const block_params = new ShortintParameters(ShortintParametersName.PARAM_SMALL_MESSAGE_2_CARRY_2_COMPACT_PK);
    // ....
}
```

## Compiling the WASM API

> tfhe-rss Makefile contains target to build the 3 possible variants of 

Reviewer: ```suggestion
tfhe-rs repo has a Makefile that contains target for each of the 3 possible variants of 
```

the WASM API:

- `make build_node_js_api` to build the nodejs API
- `make build_web_js_api` to build the browser API
- `make build_web_js_api_parallel` to build the browser API with parallelism

The compiled WASM package will be in tfhe/pkg.

> //TODO: get NPM packet

Reviewer: ```suggestion
```
